src/VisualStudio/Core/Def/Implementation/Snippets/AbstractSnippetExpansionClient.cs

```diff
@@ -393,7 +474,361 @@ public virtual bool TryInsertExpansion(int startPositionInSubjectBuffer, int end
+        /// <param name="method">The currently-selected method in Signature Help.</param>
+        /// <param name="cancellationToken">A cancellation token the operation may observe.</param>
+        public void MoveToSpecificMethod(IMethodSymbol method, CancellationToken cancellationToken)
+        {
+            AssertIsForeground();
+
+            if (ExpansionSession is null)
+            {
+                return;
+            }
+
+            if (SymbolEqualityComparer.Default.Equals(_state._method, method))
+            {
+                return;
+            }
+
+            var symbolName = _state._method?.Name ?? _state._methods.FirstOrDefault()?.Name;
+            if (symbolName != method.Name)
+            {
+                // Signature Help is showing a signature that wasn't part of the set this argument value completion
+                // session was created from. It's unclear how this state should be handled, so we stop processing
+                // Signature Help updates for the current session.
+                // TODO: https://github.com/dotnet/roslyn/issues/50636
+                _state.ClearSymbolInformation();
+                return;
+            }
+
+            if (_state._methods.IsDefaultOrEmpty)
+            {
+                // Signature Help is showing a set of overloads that don't match the overloads from the point where the
+                // argument completion session first started. It's unclear how this state should be handled, so we stop
+                // processing Signature Help updates for the current session.
+                // TODO: https://github.com/dotnet/roslyn/issues/50636
+                _state.ClearSymbolInformation();
+                return;
+            }
+
+            var textViewModel = TextView.TextViewModel;
+            if (textViewModel == null)
+            {
+                Debug.Assert(TextView.IsClosed);
+                return;
+            }
+
+            var buffer = EditorAdaptersFactoryService.GetBufferAdapter(textViewModel.DataBuffer);
+            if (buffer is not IVsExpansion expansion)
+            {
+                return;
+            }
+
+            // Track current argument values so input created/updated by a user is not lost when cycling through
+            // Signature Help overloads:
+            //
+            // 1. For each parameter of the method currently presented as a snippet, the value of the argument as
+            //    it appears in code.
+            // 2. Place the argument values in a map from parameter name to current value.
+            // 3. (Later) the values in the map can be read to avoid providing new values for equivalent parameters.
+            if (_state._method is not null)
+            {
+                foreach (var previousParameter in _state._method.Parameters)
+                {
+                    if (ExpansionSession.GetFieldValue(previousParameter.Name, out var previousValue) == VSConstants.S_OK)
```

Comment:
We feel confident there's no S_FALSE returns? :smile:

Reply:
➡️ The documentation does not allow for `S_FALSE` returns, and the current code worked in practice.